### PR TITLE
adds libsingular function to convert n_Q resp. n_Z number to mpz_t

### PR DESCRIPTION
**Pull request by ederc**

### CODE REVIEW

File: deps/src/coeffs.cpp [hunk 1]
```diff
@@ -33,6 +33,10 @@ void singular_define_coeffs(jlcxx::Module & Singular)
     Singular.method("nCoeff_has_simple_Alloc",
                     [](coeffs x) { return nCoeff_has_simple_Alloc(x) > 0; });
 
+    Singular.method("n_GetMPZ_internal", [](void * ptr, snumber *n, coeffs x) {
+        n_MPZ(reinterpret_cast<__mpz_struct *>(ptr), reinterpret_cast<number&>(n), x);
```
Comment (sebasguts) [inline deps/src/coeffs.cpp]: I think you can just pass `n` as `number` and skip the cast.

**Comment by sebasguts** [comment]
BTW, a test case would be nice.

**Comment by ederc** [comment]
Sorry, forgot to remove the cast after using number type for parameter. Should be done now.

**Comment by wbhart** [comment]
I'm not sure I would have bothered writing tests for low level functionality in libsingular, as no one is going to write tests for every other function. Rather, I would have replaced the code in Florian Heiderich's recent PR on conversion from n_Z to BigInt with a call to your new code. That already has a test function.

Of course you have to be careful to make a copy where necessary.

**Comment by ederc** [comment]
Let's just cancel this stuff, I haven't seen Florian's PR, my fault.

**Comment by wbhart** [comment]
Hmm, I actually thought you had done this the right way, as opposed to Florian's hack. At least you are using a supported interface in Singular. But it's up to you. I don't know whether you just wanted a conversion function, or if you were trying to solve a more general problem...

**Pull request closed without merging**